pkg/jsonschema/generate_config_docs.go

```diff
@@ -0,0 +1,239 @@
+package jsonschema
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/iancoleman/orderedmap"
+	"github.com/jesseduffield/lazycore/pkg/utils"
+	"github.com/samber/lo"
+
+	"gopkg.in/yaml.v3"
+)
+
+type Node struct {
+	Name        string
+	Description string
+	Default     any
+	Children    []*Node
+}
+
+const IndentLevel = 2
+
+func setComment(yamlNode *yaml.Node, description string) {
+	// If the description is more than one line, put it before the key,
+	// otherwise add it after the value on the same line.
+	if strings.Count(description, "\n") > 0 {
+		// Workaround for the way yaml formats the HeadComment if it contains
+		// blank lines: it renders these without a leading "#", but we want a
+		// leading "#" even on blank lines. However, yaml respects it if the
+		// HeadComment already contains a leading "#", so we prefix all lines
+		// (including blank ones) with "#".
+		yamlNode.HeadComment = strings.Join(
+			lo.Map(strings.Split(description, "\n"), func(s string, _ int) string {
+				if s == "" {
+					return "#" // avoid trailing space on blank lines
+				}
+				return "# " + s
+			}),
+			"\n")
+	} else {
+		yamlNode.LineComment = description
+	}
+}
+
+func (n *Node) MarshalYAML() (interface{}, error) {
+	node := yaml.Node{
+		Kind: yaml.MappingNode,
+	}
+
+	keyNode := yaml.Node{
+		Kind:  yaml.ScalarNode,
+		Value: n.Name,
+	}
+	if n.Description != "" {
+		setComment(&keyNode, n.Description)
+	}
+
+	if n.Default != nil {
+		valueNode := yaml.Node{
+			Kind: yaml.ScalarNode,
+		}
+		err := valueNode.Encode(n.Default)
+		if err != nil {
+			return nil, err
+		}
+		node.Content = append(node.Content, &keyNode, &valueNode)
+	} else if len(n.Children) > 0 {
+		childrenNode := yaml.Node{
+			Kind: yaml.MappingNode,
+		}
+		for _, child := range n.Children {
+			childYaml, err := child.MarshalYAML()
+			if err != nil {
+				return nil, err
+			}
+
+			childKey := yaml.Node{
+				Kind:  yaml.ScalarNode,
+				Value: child.Name,
+			}
+			if child.Description != "" {
```

Comment:
It would be nice if we could also separate keys that have comments with newlines, but I'm not sure how to do that after hacking it at for a couple minutes

Reply:
It doesn't seem to be possible: https://github.com/go-yaml/yaml/issues/627

But also, since almost all keys (except for keybindings) have comments now, wouldn't that consume too much vertical space?

Reply:
I'm not concerned about vertical space, but seems it's a moot point if the library doesn't support it

Reply:
But we can insert the blank lines ourselves after marshalling; it's not that hard actually, and it does indeed make things much more readable:
```diff
diff --git a/pkg/jsonschema/generate_config_docs.go b/pkg/jsonschema/generate_config_docs.go
index 000aad53b..965477f2c 100644
--- a/pkg/jsonschema/generate_config_docs.go
+++ b/pkg/jsonschema/generate_config_docs.go
@@ -144,6 +144,30 @@ func parseNode(parent *Node, name string, value *orderedmap.OrderedMap) {
 	}
 }
 
+func insertBlankLines(buffer bytes.Buffer) bytes.Buffer {
+	lines := strings.Split(strings.TrimRight(buffer.String(), "\n"), "\n")
+
+	var newBuffer bytes.Buffer
+
+	previousIndent := -1
+	wasComment := false
+
+	for _, line := range lines {
+		trimmedLine := strings.TrimLeft(line, " ")
+		indent := len(line) - len(trimmedLine)
+		isComment := strings.HasPrefix(trimmedLine, "#")
+		if isComment && !wasComment && indent <= previousIndent {
+			newBuffer.WriteString("\n")
+		}
+		newBuffer.WriteString(line)
+		newBuffer.WriteString("\n")
+		previousIndent = indent
+		wasComment = isComment
+	}
+
+	return newBuffer
+}
+
 func writeToConfigDocs(buffer *bytes.Buffer) {
 	// Remove all `---` lines
 	strData := buffer.String()
@@ -235,5 +259,7 @@ func GenerateConfigDocs() {
 		}
 	}
 
+	buffer = insertBlankLines(buffer)
+
 	writeToConfigDocs(&buffer)
 }
```

Reply:
Ah nice, I'm happy with that approach